### ISSUE TITLE
Reproduce results on SUNRGBD

### ISSUE THREAD

**Issue by junjie18** [facebookresearch/3detr#4]
Hi, thanks for you great work!

I have met problems to reproduce the results on `SUNRGBD`, the model seems not to converge after the 1st epoch. 

 I follow this [instruction](https://github.com/facebookresearch/votenet/tree/main/sunrgbd) to preprocess the dataset. I run `python sunrgbd_data.py --viz` and use MeshLab to view the data, the result seems fine. My `PYTORCH` version is 1.9.0 and `CUDA` version is 10.2

Under single gpu condition, I simply run `scripts/sunrgbd_quick.sh` and just modify the batch size to 4(I don't have V100) and the network doesn't converge. I think it's because the batch size is too small. So I tried to use 8 gpus to train the model. (Use SyncBN and scale the learning rate to 4x since the truly batch size is 32)  . But I found a problem that after several iteration the training process will hang up.(In my experiments, 104 iter at epoch 0). The reason of this phenomenon is that some parameters(for `DistributedDataParallel`) or inputs(for `SyncBN`) can't reduce gradient among all gpus. （Some gpus have gradients and some not. For example, 1st `SyncBN` of center head on gpu 0 can't get gradient_input）. I solve this problem by applying a small trick like:
`for p in model.parameters(): loss += p.sum() * 0.`

But after enlarge the batchsize to 32 and the model still doesn't converge(First epoch loss 35->17 and never drop). Is there any problem in my experiments.

By the way, how many gpus do you use in your experiments to train 1080 epoch. My training iter time is about 1.8. That means training 90 epochs will cost about 7 hours... I have install the Cythonized implementation of gIOU follow your instruction.

**Comment by imisra** [comment]
Hello,

Thanks for your interest! 
Just to make sure that your dataset is pre-processed correctly, did you try running one of the pretrained checkpoints?

If you increase the batchsize, I found it helpful to (1) scale the learning rate by the sqrt of the batchsize rather than linearly; (2) double the number of warmup epochs from 9 to 18.

For the model with 1080 epochs, we trained it using 1 GPU (released config, reduce frequency of evaluations to 90 epochs for speed up) or 8 GPUs (warmup=18 epochs, base_lr=1.97e-3, batchsize=64).

**Comment by junjie18** [comment]
Hi, thanks for your answer.
The test result of `SUNRGBD` 1080ep pretrained model is:

> Test model; Metrics mAP0.25, mAP0.50: 57.68, 29.08
> AR0.25, AR0.50: 79.14, 45.08

I still have following problems. 

- Is my bug in multi-gpus mentioned above a common state
- Can the model converge when using (batch size 4 single gpu) in your experiments
- how long will take to train 1080ep

Wish for your reply

**Comment by imisra** [comment]
Thanks for trying out the changes
- I have not seen this bug. I'll inspect more, but the solution you outlined seems to be fix for the time being
- It takes about 34-36 hours on 8 GPUs (I do infrequent testing by setting --eval_every_epoch to 500)
